src/Rematch.jl

```diff
@@ -31,6 +39,9 @@ function handle_destruct_fields(value::Symbol, pattern, subpatterns, len, get::S
             push!(fields, (:($i:($len-$(length(subpatterns)-i))), subpattern.args[1]))
         elseif seen_splat
             push!(fields, (:($len-$(length(subpatterns)-i)), subpattern))
+        elseif (subpattern isa Expr) && (subpattern.head == :kw)
+            T_sym = Meta.quot(subpattern.args[1])
+            push!(fields, (:($T_sym), subpattern.args[2]))
```

Comment:
nit: isn't the lhs here the name of the field, not the name of the type? Maybe I'm misreading this.
If so, could we name it `field_sym` or `fieldname` or something instead of `T_sym`?